File: src/main/java/com/kishlaly/ta/analyze/tasks/ThreeDisplays.java
package com.kishlaly.ta.analyze.tasks;

/**
 * Named after Alex Elder's famous three-displays-technique
 * however, here is used only Longterm and Midterm
 * <p>
 * Indicators:
 * EMA26 (close) on the first screen
 * MACD (12 26 9 close) on the second screen
 * EMA13 (close) on the second screen
 * STOCH (14 1 3 close) on the second screen
 */
public class ThreeDisplays extends AbstractTask {

    public static class Config {

        // 3 gives fewer signals, but they are more reliable
        public static int NUMBER_OF_EMA26_VALUES_TO_CHECK = 4;

        public static int STOCH_OVERSOLD = 40;
        public static int STOCH_OVERBOUGHT = 70;
        public static int STOCH_VALUES_TO_CHECK = 5;

        // in percent from the middle to the top of the channel
        public static int FILTER_BY_KELTNER = 20;

        // filtering signals if a quote closed above FILTER_BY_KELTNER
        // tests show better results when this check is turned off
        public static boolean FILTER_BY_KELTNER_ENABLED = true;

        // configurable indicator values
        public static int STOCH_CUSTOM = -1;
        public static int EMA26_TOTAL_BARS_CHECK = -1;
        public static int EMA26_ABOVE_BARS = -1;
        public static int BOLLINGER_TOTAL_BARS_CHECK = -1;
        public static int BOLLINGER_CROSSED_BOTTOM_BARS = -1;
        public static int QUOTE_FROM_END_TO_USE = -1;
    }

}

File: src/main/kotlin/com/kishlaly/ta/analyze/tasks/ThreeDisplays.kt
package com.kishlaly.ta.analyze.tasks

class ThreeDisplays : AbstractTask() {

    object Config {
        // 3 gives fewer signals, but they are more reliable
        var NUMBER_OF_EMA26_VALUES_TO_CHECK = 4
        var STOCH_OVERSOLD = 40
        var STOCH_OVERBOUGHT = 70
        var STOCH_VALUES_TO_CHECK = 5

        // in percent from the middle to the top of the channel
        var FILTER_BY_KELTNER = 50

        // filtering signals if a quote closed above FILTER_BY_KELTNER
        // tests show better results when this check is turned off
        var FILTER_BY_KELTNER_ENABLED = true

        // configurable indicator values
        var STOCH_CUSTOM = 30
        var EMA26_TOTAL_BARS_CHECK = 7
        var EMA26_ABOVE_BARS = 4
        var BOLLINGER_TOTAL_BARS_CHECK = 3
        var BOLLINGER_CROSSED_BOTTOM_BARS = 1
        var QUOTE_FROM_END_TO_USE = 2
    }

}